packages/node/src/lib/retry.spec.ts

```diff
@@ -0,0 +1,205 @@
+import nock from 'nock';
+import { Novu } from '../index';
+
+const BACKEND_URL = 'http://example.com';
+const TOPICS_PATH = '/v1/topics';
+const TRIGGER_PATH = '/v1/events/trigger';
+
+jest.setTimeout(15000);
+
+const allEqual = (arr: Array<string>) => arr.every((val) => val === arr[0]);
+
+class NetworkError extends Error {
+  constructor(public code: string) {
+    super('Network Error');
+  }
+}
+
+describe('Novu Node.js package - Retries and idempotency-key', () => {
+  afterEach(() => {
+    nock.cleanAll();
+    nock.enableNetConnect();
+  });
+
+  const novu = new Novu('fake-key', {
+    backendUrl: BACKEND_URL,
+    retryConfig: {
+      retryMax: 3,
+      waitMax: 1,
+      waitMin: 1,
+    },
+  });
+
+  it('should retry trigger and generate idempotency-key only once for request', async () => {
+    const idempotencyKeys: string[] = [];
+
+    nock(BACKEND_URL)
+      .post(TRIGGER_PATH)
+      .times(3)
+      .reply(function (_url, _body) {
+        idempotencyKeys.push(this.req.getHeader('idempotency-key') as string);
+
+        return [500, { message: 'Server Exception' }];
+      });
+
+    nock(BACKEND_URL)
+      .post(TRIGGER_PATH)
+      .reply(201, { acknowledged: true, transactionId: '1003' });
+
+    const result = await novu.trigger('fake-workflow', {
+      to: { subscriberId: '123' },
+      payload: {},
+    });
+
+    // all idempotency keys are supposed to be same.
+    expect(allEqual(idempotencyKeys)).toBeTruthy();
+    expect(result.status).toEqual(201);
+    expect(result.request.headers['idempotency-key']).toBeDefined();
+  });
+
+  it('should generate different idempotency-key for each request', async () => {
+    nock(BACKEND_URL)
+      .post(TRIGGER_PATH)
+      .reply(500, { message: 'Server Exception' });
+
+    nock(BACKEND_URL)
+      .post(TRIGGER_PATH)
+      .times(10)
+      .reply(201, { acknowledged: true, transactionId: '1003' });
+
+    const idempotencyKeys: string[] = [];
+
+    for (let i = 0; i < 10; i++) {
+      const result = await novu.trigger('fake-workflow', {
+        to: { subscriberId: '123' },
+        payload: {},
+      });
+
+      idempotencyKeys.push(result.request?.headers['idempotency-key']);
+    }
+
+    expect(allEqual(idempotencyKeys)).toEqual(false);
+  });
```

Comment:
new test case - I want to make sure that idempotency key is different between **requests**